### PR TITLE
Snap authorship modal

### LINKED ISSUE #21888: Header metadata modal
Let's display some metadata (including version, description, source, etc) [within a modal](https://www.figma.com/file/Wlwk0cTx6PQkNhiX7vN13e/Snaps-Platform-v1?type=design&node-id=1136%3A14287&mode=design&t=rViXz6WgGhxmuAcg-1) opened from the authorship header

### PR DESCRIPTION
**Pull request by GuillaumeRx**
## **Description**

This adds a modal to the `SnapAuthorshipHeader` containing the snap infos

## **Related issues**

Fixes: #21888 

## **Manual testing steps**

1. Try installing a snap
2. On the install screen, click the info icon on the right of the authorship header
3. Also works on the other screens showing the authorship header

## **Screenshots/Recordings**
image

image

image

image

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/confirm/shared/info/row/row.tsx [hunk 1]
```diff
@@ -67,6 +67,7 @@ export const ConfirmInfoRow = ({
 }: ConfirmInfoRowProps) => (
   <ConfirmInfoRowContext.Provider value={{ variant }}>
     <Box
+      className="confirm-info-row"
```
Comment (david0xd) [inline ui/components/app/confirm/shared/info/row/row.tsx:70]: Is this `className` used for something?

File: app/_locales/en/messages.json [hunk 2]
```diff
@@ -2426,6 +2426,10 @@
   "metaMaskConnectStatusParagraphTwo": {
     "message": "The connection status button shows if the website you’re visiting is connected to your currently selected account."
   },
+  "metadataModalSourceTooltip": {
+    "message": "$1 is hosted on npm and $2 is this Snap’s unique unique identifier.",
```
Comment (david0xd) [inline app/_locales/en/messages.json:2430]: Why is word `unique` added here two times?

**Review by david0xd — DISMISSED** [review]

**Review by david0xd — DISMISSED** [review]

**Review by david0xd — DISMISSED** [review]

**Review by david0xd — DISMISSED** [review]

**Review by FrederikBolding — CHANGES_REQUESTED** [review]
image

Found a scrolling problem.

File: ui/components/app/snaps/snap-metadata-modal/snap-metadata-modal.js [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+  const { website = undefined } = snapRegistryData?.metadata ?? {};
+  const [safeWebsite, setSafeWebsite] = useState(null);
+
+  useEffect(() => {
+    const performPhishingCheck = async () => {
+      const phishingResult = await getPhishingResult(website);
+
+      if (!phishingResult.result) {
+        setSafeWebsite(new URL(website));
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-metadata-modal/snap-metadata-modal.js:80]: Can we move this to a shared hook between the modal and snaps settings?

Feels like there is a lot of duplication between those two components?

Comment (GuillaumeRx) [reply ui/components/app/snaps/snap-metadata-modal/snap-metadata-modal.js]: implemented in cd5734f

**Review by FrederikBolding — APPROVED** [review]

**Review by david0xd — APPROVED** [review]

**Pull request merged**